backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala

```diff
@@ -917,43 +917,106 @@ trait StandardAsyncExecutionActor
-          (retryWithMoreMemory, memoryRetryFactor) match {
-            case (true, Some(multiplier)) => refineV[GreaterEqualOne](currentMemoryMultiplier.value * multiplier.value) match {
-              case Left(_) => FailedNonRetryableExecutionHandle(MemoryMultiplierNotPositive(jobDescriptor.key.tag, None, currentMemoryMultiplier, multiplier), failed.returnCode)
-              case Right(newMultiplier) => FailedRetryableExecutionHandle(failed.throwable, failed.returnCode, newMultiplier)
+      case failedRetryableOrNonRetryable: FailedExecutionHandle =>
+
+        val kvsFromPreviousAttempt = jobDescriptor.prefetchedKvStoreEntries.collect {
+          case (key: String, kvPair: KvPair) => key -> kvPair
+        }
+        val kvsForNextAttempt = failedRetryableOrNonRetryable.kvPairsToSave match {
+          case Some(kvPairs) => kvPairs.map {
+            case kvPair@KvPair(ScopedKey(_, _, key), _) => key -> kvPair
+          }.toMap
+          case None => Map.empty[String, KvPair]
+        }
+
+        val maxRetriesNotReachedYet = previousFailedRetries < maxRetries
+        failedRetryableOrNonRetryable match {
+          case failed: FailedNonRetryableExecutionHandle if maxRetriesNotReachedYet =>
+            val currentMemoryMultiplier = jobDescriptor.key.memoryMultiplier
+            (retryWithMoreMemory, memoryRetryFactor) match {
+              case (true, Some(multiplier)) => refineV[GreaterEqualOne](currentMemoryMultiplier.value * multiplier.value) match {
+                case Left(_) => Future.successful(
+                  FailedNonRetryableExecutionHandle(
+                    MemoryMultiplierNotPositive(jobDescriptor.key.tag, None, currentMemoryMultiplier, multiplier),
+                    failed.returnCode,
+                    None
+                  )
+                )
+                case Right(newMultiplier) => saveAttrsAndRetry(failed, kvsFromPreviousAttempt, kvsForNextAttempt, incFailedCount = true, Option(newMultiplier))
+              }
+              case (_, _) => saveAttrsAndRetry(failed, kvsFromPreviousAttempt, kvsForNextAttempt, incFailedCount = true)
             }
-            case (_, _) => FailedRetryableExecutionHandle(failed.throwable, failed.returnCode, currentMemoryMultiplier)
-          }
+          case failedNonRetryable: FailedNonRetryableExecutionHandle => Future.successful(failedNonRetryable)
+          case failedRetryable: FailedRetryableExecutionHandle => saveAttrsAndRetry(failedRetryable, kvsForNextAttempt, kvsForNextAttempt, incFailedCount = false)
         }
       case _ => backendExecutionStatus
     }
   }
 
+  private def saveAttrsAndRetry(failedExecHandle: FailedExecutionHandle,
+                        kvPrev: Map[String, KvPair],
+                        kvNext: Map[String, KvPair],
+                        incFailedCount: Boolean,
+                        memoryMultiplier: Option[GreaterEqualRefined] = None): Future[FailedRetryableExecutionHandle] = {
```

Comment:
No, since `type GreaterEqualRefined = Refined[Double, GreaterEqualOne]` and `refineMv[GreaterEqualOne](some_double_value)` returns value of type `GreaterEqualRefined`